Gradle Jar/main directory issues, 1.7.1.FInal

Just updated to the latest (1.7.1.Final) quarkus, and the tests run fine, but when running `quarkusDev` I get the following output, and no running server:

```
> Task :WebServer:Server:quarkusDev
Listening for transport dt_socket at address: 5005
Exception in thread "main" java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: java.io.UncheckedIOException: Error while reading file as JAR: /home/gstewart/GitRepos/task-timekeeper/WebServer/WebLibrary/build/resources/main
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:150)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:63)
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.io.UncheckedIOException: Error while reading file as JAR: /home/gstewart/GitRepos/task-timekeeper/WebServer/WebLibrary/build/resources/main
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:135)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:85)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:144)
        ... 1 more
Caused by: java.lang.RuntimeException: java.io.UncheckedIOException: Error while reading file as JAR: /home/gstewart/GitRepos/task-timekeeper/WebServer/WebLibrary/build/resources/main
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:374)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:51)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:132)
        ... 3 more
Caused by: java.io.UncheckedIOException: Error while reading file as JAR: /home/gstewart/GitRepos/task-timekeeper/WebServer/WebLibrary/build/resources/main
        at io.quarkus.bootstrap.classloading.JarClassPathElement.<init>(JarClassPathElement.java:46)
        at io.quarkus.bootstrap.classloading.ClassPathElement.fromPath(ClassPathElement.java:51)
        at io.quarkus.bootstrap.app.CuratedApplication.createDeploymentClassLoader(CuratedApplication.java:265)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:330)
        ... 5 more
Caused by: java.io.FileNotFoundException: /home/gstewart/GitRepos/task-timekeeper/WebServer/WebLibrary/build/resources/main (No such file or directory)
        at java.util.zip.ZipFile.open(Native Method)
        at java.util.zip.ZipFile.<init>(ZipFile.java:225)
        at java.util.zip.ZipFile.<init>(ZipFile.java:155)
        at java.util.jar.JarFile.<init>(JarFile.java:166)
        at java.util.jar.JarFile.<init>(JarFile.java:130)
        at io.smallrye.common.io.jar.JarFiles.create(JarFiles.java:34)
        at io.quarkus.bootstrap.classloading.JarClassPathElement.<init>(JarClassPathElement.java:44)
        ... 8 more
<============-> 98% EXECUTING [48s]
> :WebServer:Server:quarkusDev

```

Runs fine on `1.6.1.Final`. Let me know what else would be good to put here as far as output.

--- a/core/deployment/src/main/java/io/quarkus/deployment/dev/IDEDevModeMain.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/dev/IDEDevModeMain.java
@@ -73,14 +73,17 @@ public class IDEDevModeMain implements BiConsumer<CuratedApplication, Map<String
             sourceDirectories.add(srcDir.getPath());
             sourceParents.add(srcDir.getParent());
         }
-
+        String resourceDirectory = null;
+        if (module.getSourceSet().getResourceDirectory() != null) {
+            resourceDirectory = module.getSourceSet().getResourceDirectory().getPath();
+        }
         return new DevModeContext.ModuleInfo(key,
                 module.getArtifactCoords().getArtifactId(),
                 module.getProjectRoot().getPath(),
                 sourceDirectories,
                 QuarkusModelHelper.getClassPath(module).toAbsolutePath().toString(),
                 module.getSourceSourceSet().getResourceDirectory().toString(),
-                module.getSourceSet().getResourceDirectory().getPath(),
+                resourceDirectory,
                 sourceParents,
                 module.getBuildDir().toPath().resolve("generated-sources").toAbsolutePath().toString(),
                 module.getBuildDir().toString());

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/builder/QuarkusModelBuilder.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/builder/QuarkusModelBuilder.java
@@ -316,9 +316,13 @@ public class QuarkusModelBuilder implements ParameterizedToolingModelBuilder<Mod
     }
 
     private SourceSetImpl convert(SourceSet sourceSet) {
+        if (sourceSet.getOutput().getResourcesDir().exists()) {
+            return new SourceSetImpl(
+                    sourceSet.getOutput().getClassesDirs().getFiles(),
+                    sourceSet.getOutput().getResourcesDir());
+        }
         return new SourceSetImpl(
-                sourceSet.getOutput().getClassesDirs().getFiles(),
-                sourceSet.getOutput().getResourcesDir());
+                sourceSet.getOutput().getClassesDirs().getFiles());
     }
 
     private io.quarkus.bootstrap.resolver.model.SourceSet getSourceSourceSet(SourceSet sourceSet) {

--- a/devtools/gradle/src/test/java/io/quarkus/gradle/builder/QuarkusModelBuilderTest.java
+++ b/devtools/gradle/src/test/java/io/quarkus/gradle/builder/QuarkusModelBuilderTest.java
@@ -69,7 +69,7 @@ class QuarkusModelBuilderTest {
         assertEquals(new File(projectDir, "build"), workspaceModule.getBuildDir());
         final SourceSet sourceSet = workspaceModule.getSourceSet();
         assertNotNull(sourceSet);
-        assertEquals(new File(projectDir, "build/resources/main"), sourceSet.getResourceDirectory());
+        assertNull(sourceSet.getResourceDirectory());
         assertEquals(1, sourceSet.getSourceDirectories().size());
         assertEquals(new File(projectDir, "build/classes/java/main"), sourceSet.getSourceDirectories().iterator().next());
         final SourceSet sourceSourceSet = workspaceModule.getSourceSourceSet();

--- a/independent-projects/bootstrap/gradle-resolver/src/main/java/io/quarkus/bootstrap/resolver/model/impl/SourceSetImpl.java
+++ b/independent-projects/bootstrap/gradle-resolver/src/main/java/io/quarkus/bootstrap/resolver/model/impl/SourceSetImpl.java
@@ -10,11 +10,15 @@ import java.util.stream.Collectors;
 public class SourceSetImpl implements SourceSet, Serializable {
 
     private Set<File> sourceDirectories = new HashSet<>();
-    private final File resourceDirectory;
+    private File resourceDirectory;
 
     public SourceSetImpl(Set<File> sourceDirectories, File resourceDirectory) {
         this.sourceDirectories.addAll(sourceDirectories);
         this.resourceDirectory = resourceDirectory;
+    }
+
+    public SourceSetImpl(Set<File> sourceDirectories) {
+        this.sourceDirectories.addAll(sourceDirectories);
     }
 
     public void addSourceDirectories(Set<File> files) {

--- a/independent-projects/bootstrap/gradle-resolver/src/main/java/io/quarkus/bootstrap/util/QuarkusModelHelper.java
+++ b/independent-projects/bootstrap/gradle-resolver/src/main/java/io/quarkus/bootstrap/util/QuarkusModelHelper.java
@@ -135,8 +135,9 @@ public class QuarkusModelHelper {
             WorkspaceModule module = model.getWorkspace().getMainModule();
             module.getSourceSet().getSourceDirectories().stream().filter(File::exists).map(File::toPath)
                     .forEach(paths::add);
-            if (module.getSourceSet().getResourceDirectory().exists()) {
-                paths.add(module.getSourceSet().getResourceDirectory().toPath());
+            File resourceDirectory = module.getSourceSet().getResourceDirectory();
+            if (resourceDirectory != null && resourceDirectory.exists()) {
+                paths.add(resourceDirectory.toPath());
             }
             appArtifact.setPaths(paths.build());
         }

--- a/integration-tests/gradle/src/test/resources/basic-java-application-project/settings.gradle
+++ b/integration-tests/gradle/src/test/resources/basic-java-application-project/settings.gradle
@@ -9,6 +9,3 @@ pluginManagement {
     }
 }
 rootProject.name='code-with-quarkus'
-include 'library'
-include 'application'
-

